git_machete/tests/functional/test_machete.py

```diff
@@ -1532,10 +1541,10 @@ def test_update_with_fork_point_specified(self) -> None:
                 "specified by the option '-f' from the current branch."
         )
 
-    @mock.patch('urllib.request.urlopen', ContextManager)
-    @mock.patch('urllib.request.Request', GitAPIState)
+    @mock.patch('urllib.request.urlopen', MockContextManager)
+    @mock.patch('urllib.request.Request', MockGitAPIRequest)
     def test_retarget_pr(self) -> None:
-        GitAPIState.set_initial_values()
+        MockGitAPIRequest.git_api_state.set_initial_values()
```

Comment:
It's definitely cleaner now after the split of MockRequest/MockAPIState... but still, there's a **global mutable state** here, shared b/w all test cases :/

Reply:
Has there been any obstacle in following the path suggested inhttps://github.com/VirtusLab/git-machete/pull/214/files/674de5bc22d1ab98ef84227514ea9e7edb17438f..de3711b32d3d14f93485aa6bfdaa4c11c70b7b89#r705527546 ? maybe `new_callable` (instead of `new`) param instead of `mock.patch` should be used? 🤔

Reply:
(I've introduced some minor fixes to the snippet with suggested new layout)

Reply:
To recap: we want **each test case to have its own `MockGitAPIState`** that is NOT shared with other test cases.